extensions/ql-vscode/src/data-extensions-editor/auto-model-codeml-queries.ts

```diff
@@ -0,0 +1,223 @@
+import { CodeQLCliServer, SourceInfo } from "../codeql-cli/cli";
+import { QueryRunner } from "../query-server";
+import { DatabaseItem } from "../databases/local-databases";
+import { ProgressCallback } from "../common/vscode/progress";
+import * as Sarif from "sarif";
+import { qlpackOfDatabase, resolveQueries } from "../local-queries";
+import { extLogger } from "../common/logging/vscode";
+import { Mode } from "./shared/mode";
+import { QlPacksForLanguage } from "../databases/qlpack";
+import { createLockFileForStandardQuery } from "../local-queries/standard-queries";
+import { CancellationToken, CancellationTokenSource } from "vscode";
+import { getOnDiskWorkspaceFolders } from "../common/vscode/workspace-folders";
+import { showAndLogExceptionWithTelemetry, TeeLogger } from "../common/logging";
+import { QueryResultType } from "../query-server/new-messages";
+import { telemetryListener } from "../common/vscode/telemetry";
+import { redactableError } from "../common/errors";
+import { interpretResultsSarif } from "../query-results";
+import { join } from "path";
+import { assertNever } from "../common/helpers-pure";
+
+type Options = {
```

Comment:
I don't quite get why we're calling things "Options" - to me they look like normal method inputs. Also it's hard to know the difference between `Options` and `AutoModelQueryOptions`

Reply:
These are regular method inputs, but with this many inputs it becomes somewhat harder to reason about the order of the arguments. By using an object, they are at least all named so it becomes easier to reason about them.

I've renamed it to `AutoModelQueriesOptions` and moved `mode` into it. I've also moved it closer to the actual function definition so it's easier to distinguish between the two options types.

Reply:
Thanks for the changes! The term "options" still seems a bit off to me because it makes me think of optional or variadic arguments, but I don't have a good suggestions atm (only "inputs" comes to mind).